### PR TITLE
Add generic error message for send operation

### PR DESCRIPTION
**Pull request by OKendigelyan**
## Proposed changes

This PR replaces detailed error messages with generic ones to hide potential sensitive information and make it more user friendly.

### CODE REVIEW

**Review by asiia-trilitech — COMMENTED** [review]
@OKendigelyan just to doublecheck - do we need to add any changes to the desktop?

File: apps/web/src/views/Tokens/SendTokensFormPage.tsx
Comment (asiia-trilitech) [inline apps/web/src/views/Tokens/SendTokensFormPage.tsx:1]: Why these files are being deleted? Are they not used across the codebase?

https://github.com/trilitech/umami-v2/blob/main/apps/web/src/views/Tokens/Token.tsx#L70

File: packages/core/src/estimate.ts [hunk 1]
```diff
@@ -59,13 +57,14 @@ const estimateToEstimation = (estimate: Estimate): Estimation => ({
...
-export const handleTezError = (err: Error): string => {
+export const handleTezError = (err: Error): string | undefined => {
   if (err.message.includes("subtraction_underflow")) {
     return "Insufficient balance, please make sure you have enough funds.";
   } else if (err.message.includes("contract.non_existing_contract")) {
     return "Contract does not exist, please check if the correct network is selected.";
   } else if (err.message.includes("staking_to_delegate_that_refuses_external_staking")) {
     return "The baker you are trying to stake to does not accept external staking.";
+  } else if (err.message.includes("empty_implicit_delegated_contract")) {
+    return "Emptying an implicit delegated account is not allowed.";
```
Comment (asiia-trilitech) [inline packages/core/src/estimate.ts:68]: nit: maybe something like 

`Emptying an implicit delegated account is not allowed. End delegation before ?trying? again.`

**Review by asiia-trilitech — CHANGES_REQUESTED** [review]
Thank you @OKendigelyan! 
PR looks good to me - please check the comments & I'm happy to approve after!

File: apps/web/src/views/Tokens/SendTokensFormPage.tsx
Comment (OKendigelyan) [reply apps/web/src/views/Tokens/SendTokensFormPage.tsx:1]: they've been absolete

File: apps/web/src/components/Menu/ErrorLogsMenu/ErrorLogsMenu.tsx [hunk 2]
```diff
@@ -58,6 +58,11 @@ export const ErrorLogsMenu = () => {
                     <Text color={color("700")} size="sm">
                       {errorLog.timestamp}
                     </Text>
+                    {errorLog.technicalDetails && (
+                      <Text marginTop="12px" color={color("700")} size="sm">
+                        {errorLog.technicalDetails}
```
Comment (ajinkyaraj-23) [inline apps/web/src/components/Menu/ErrorLogsMenu/ErrorLogsMenu.tsx:63]: isnt technicalDetails same as description(old version) if it is visible in Error logs that would be counterproductive to this security issue.

Comment (OKendigelyan) [reply apps/web/src/components/Menu/ErrorLogsMenu/ErrorLogsMenu.tsx:63]: assuming we'll require a password from the user to show errors page it should be quite secure
generic message will be shown in the toast

Comment (ajinkyaraj-23) [reply apps/web/src/components/Menu/ErrorLogsMenu/ErrorLogsMenu.tsx:63]: The issue was not of security but exposing the detailed architecture of the app. I believe this should be removed and only description and error code should be shown to the user. What do you think @asiia-trilitech ?

File: packages/core/src/estimate.ts (hunk 1 above)
Comment (ajinkyaraj-23) [reply packages/core/src/estimate.ts]: Changed. Thanks

File: apps/web/src/components/Menu/ErrorLogsMenu/ErrorLogsMenu.tsx (hunk 2 above)
Comment (ajinkyaraj-23) [reply apps/web/src/components/Menu/ErrorLogsMenu/ErrorLogsMenu.tsx]: Removed for now. We can expand on handleTezErrors going forwards to cover more types of errors.

**Comment by ajinkyaraj-23** [comment]
Created an issue for "too generic error msgs for input validation/secret key validation in restore secret key form". Can be addressed later.

**Review by ajinkyaraj-23 — APPROVED** [review]
LGTM

**Comment by OKendigelyan** [comment]
> [ajinkyaraj-23](/ajinkyaraj-23) merged commit [`b3ef307`](/trilitech/umami-v2/commit/b3ef307246e3ebfeb51d8cc2c1d35489d88b6028) into main [Dec 6, 2024](https://github.com/trilitech/umami-v2/pull/2185#event-15560223264)

fixed [here](https://github.com/trilitech/umami-v2/pull/2210)

**Pull request merged**